### PR TITLE
New Send Screen Flow

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/screens/send/components/enter-amount/enter-amount.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import stylesProvider from './styles';
+import { withTheme, IThemeProps } from '../../../../core/theme/with-theme';
+import { Text } from '../../../../library';
+import { smartConnect } from '../../../../core/utils/smart-connect';
+import { translate } from '../../../../core/i18n';
+import { Amount } from '../../../../components/amount/amount';
+import { Blockchain } from '../../../../core/blockchain/types';
+import { ITokenConfig } from '../../../../core/blockchain/types/token';
+
+export interface IExternalProps {
```
Comment (krisboit) [inline src/screens/send/components/enter-amount/enter-amount.tsx]: the interface of the component is too complicated, what about
```ts
availableAmount: string, // amount excepting fees
blockchain: Blockchain,
token: ITokenConfig,
value: string,
onChange: (value: string) => any
```

File: src/screens/send/components/header-step-by-step/header-step-by-step.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { View, Dimensions, TouchableOpacity } from 'react-native';
+import stylesProvider from './styles';
+import { withTheme, IThemeProps } from '../../../../core/theme/with-theme';
+import { Text } from '../../../../library';
+import { smartConnect } from '../../../../core/utils/smart-connect';
+import { BASE_DIMENSION } from '../../../../styles/dimensions';
+
+export interface IExternalProps {
+    steps: { title: string; selected?: boolean }[];
+    selectPreviousStep: (index: number) => void;
```
Comment (georgeburduhos) [inline src/screens/send/components/header-step-by-step/header-step-by-step.tsx]: too complicated with selectPreviousStep
Make it easier. pass a list of steps with selected 0 or 1 - one step will be selected
Example:
[0, 0, 0, 1, 0] - mark all steps (visually) as selected until you reach the 1 value, all after 1 are not selectee

File: src/screens/send/send.tsx [hunk 3]
```diff
@@ -427,193 +427,291 @@ export class SendScreenComponent extends React.Component<
...
-                </ScrollView>
+                        ]}
+                        selectPreviousStep={(index: number) => {
+                            if (index === 1 && isStep3) {
+                                this.setState({ isStep2: true, isStep3: false });
+                            } else if (index === 0) {
+                                this.setState({ isStep2: false, isStep3: false });
+                            }
+                        }}
+                    />
```
Comment (georgeburduhos) [inline src/screens/send/send.tsx]: It makes sense here but I would've added and extra component like
SelectToAddressComponent - that is basically the step one for send and will have: Address book, Transfer between account component, address input, qrcode functionality
This would make Send screen just a handler for each step

File: src/core/i18n/translation/locales/en.ts [hunk 4]
```diff
@@ -189,7 +183,10 @@ export const translation: ITranslations = {
             alertEditDescription: 'Use a relevant name or an alias for this address',
             deleteContact: 'Delete contact',
             editContactName: 'Edit name',
-            reviewTransaction: 'Review transaction on your hardware wallet'
+            reviewTransaction: 'Review transaction on your hardware wallet',
+            addAddress: 'Add\naddress',
```
[the remark this answers is not in the document]
Comment (NovacAndrei) [reply src/core/i18n/translation/locales/en.ts]: `Send` it's disabled on extension

File: src/components/password-modal/password-modal.tsx [hunk 5]
```diff
@@ -104,9 +104,7 @@ export class PasswordModalComponent extends React.Component<
                 this.passwordRequestDeferred &&
                     this.passwordRequestDeferred.resolve(keychainPassword.password);
 
-                if (this.props.onPassword) {
-                    this.props.onPassword(keychainPassword.password);
-                }
+                this.props.onPassword && this.props.onPassword(keychainPassword.password);
```
Comment (georgeburduhos) [inline src/components/password-modal/password-modal.tsx]: why the changes here? - there is another PR for this

File: src/components/password-modal/password-modal.tsx [hunk 6]
```diff
@@ -156,10 +154,8 @@ export class PasswordModalComponent extends React.Component<
                 this.passwordRequestDeferred && this.passwordRequestDeferred.resolve(value);
             }
 
-            this.setState({
-                visible: false
-            });
-            this.props.onPassword(value);
+            this.setState({ visible: false });
```
Comment (georgeburduhos) [inline src/components/password-modal/password-modal.tsx]: same here

File: src/screens/send/components/add-address/add-address.tsx [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+    ChainIdType,
+    Blockchain
+} from '../../../../core/blockchain/types';
+import { addContact } from '../../../../redux/contacts/actions';
+import { connect } from 'react-redux';
+import { QrModalReader } from '../../../../components/qr-modal/qr-modal';
+
+export interface IExternalProps {
+    account: IAccountState;
+    accounts: IAccountState[];
```
Comment (georgeburduhos) [inline src/screens/send/components/add-address/add-address.tsx]: accounts can be taken from redux directly, and remove from send screen.
If if other redux props can be move here only

File: src/screens/send/send.tsx [hunk 8]
```diff
@@ -10,14 +9,10 @@ import { connect } from 'react-redux';
 import { Text } from '../../library';
 import { translate } from '../../core/i18n';
 import { getBlockchain } from '../../core/blockchain/blockchain-factory';
-import { QrModalReader } from '../../components/qr-modal/qr-modal';
 import { withNavigationParams, INavigationProps } from '../../navigation/with-navigation-params';
 import { addContact } from '../../redux/contacts/actions';
-import { IContactState, IContactsState } from '../../redux/contacts/state';
+import { IContactsState } from '../../redux/contacts/state';
```
Comment (georgeburduhos) [inline src/screens/send/send.tsx]: you can move contacts to add-address
